### ISSUE TITLE
Workflows should be ignoreable

### ISSUE THREAD

**Issue by burgerdev** [ilastik/ilastik#1024]
As far as I can tell we have two workflow types (object classification and tracking) which have parent workflows for shared functionality and child workflows for the actual implementations (from segmentation, from prediction, ...). In the project selection screen only the child workflows are shown, which is good, but the implementation of it is hacky: https://github.com/ilastik/ilastik/blob/master/ilastik/workflow.py#L219. There should be a documented way of doing this, e.g. a property `Workflow.isValid`.

**Comment by akreshuk** [comment]
@stuarteberg , is it still that bad?

**Comment by stuarteberg** [comment]
Yes, it's [still a bit of a hack](https://github.com/ilastik/ilastik/blob/e9a3b7218ff121eb171f6e3687f8c5ec44396be0/ilastik/workflow.py#L262).
